### PR TITLE
Avoid (temporary?) integer overflow

### PR DESCRIPTION
**Pull request by eustas**
Drive-by: simplify division by 2:
https://godbolt.org/z/sq91EMKT8

### CODE REVIEW

**Comment by veluca93** [comment]
@jonsneyers to double-check if this simplification can be applied to the SIMD code too

File: lib/jxl/modular/transform/squeeze.cc [hunk 1]
```diff
@@ -119,8 +119,7 @@ Status InvHSqueeze(Image &input, uint32_t c, uint32_t rc, ThreadPool *pool) {
       pixel_type left = (x ? p_out[(x << 1) - 1] : avg);
       pixel_type tendency = SmoothTendency(left, avg, next_avg);
       pixel_type diff = diff_minus_tendency + tendency;
-      pixel_type A =
-          ((avg * 2) + diff + (diff > 0 ? -(diff & 1) : (diff & 1))) >> 1;
+      pixel_type A = avg + (diff / 2);
```
Comment (jonsneyers) [inline lib/jxl/modular/transform/squeeze.cc:122]: There's a similar line in InvVSqueeze, can do the same simplification there.

Comment (eustas) [reply lib/jxl/modular/transform/squeeze.cc:122]: Sure. One moment.

**Comment by jonsneyers** [comment]
Wow, good find! How did I ever end up with that complicated expression if it just boils down to `avg + diff/2`?

**Comment by eustas** [comment]
I did that in a thought experiment. Both addends (`avg * 2` and "fixed" diff) have the rightmost bit set to 0 -> their sum have the rightmost bit set to 0 as well -> likely no surprises. But going to double-check soon.

**Comment by jonsneyers** [comment]
We should also clean this up in the spec, and probably the SIMD code can be made slightly faster by doing the simplification also there.

**Pull request merged**